src/lib/connector.ts

```diff
@@ -1,64 +1,44 @@
-import { Signer, utils, ethers } from "ethers";
-import {
-  ConnectionOptions,
-  Connection,
-  Token,
-  SupportedNetwork,
-} from "../interfaces.js";
+import { Signer, ethers } from "ethers";
+import { ConnectionOptions, Connection, Token } from "../interfaces.js";
 import { list } from "./list.js";
 import { createToken } from "./token.js";
-import { query } from "./query.js";
+import { read, write } from "./query.js";
 import { create } from "./create.js";
 import { hash } from "./hash.js";
+import { receipt } from "./tableland-calls.js";
+import { SUPPORTED_NETWORKS, contractAddresses } from "./util.js";
 
 declare let globalThis: any;
 
-const SUPPORTED_NETWORKS: SupportedNetwork[] = [
-  {
-    key: "rinkeby",
-    phrase: "Ethereum Rinkeby",
-  },
-];
-
 export async function getSigner(): Promise<Signer> {
   await globalThis.ethereum.request({ method: "eth_requestAccounts" });
   const provider = new ethers.providers.Web3Provider(globalThis.ethereum);
   const signer = provider.getSigner();
   return signer;
 }
 
-export async function userCreatesToken(signer: Signer): Promise<Token> {
-  const sign = {
-    signMessage: async (message: Uint8Array): Promise<Uint8Array> => {
-      const sig = await signer.signMessage(message);
-      return utils.arrayify(sig);
-    },
-  };
-  const iat = ~~(Date.now() / 1000);
-  const exp = iat + 60 * 60 * 10; // Default to ~10 hours
-
-  // WARN: This is a non-standard JWT
-  // Borrows ideas from: https://github.com/ethereum/EIPs/issues/1341
-  const iss = await signer.getAddress();
-  const network = await signer.provider?.getNetwork();
-  const chain = network?.chainId ?? "unknown";
-  let net = network?.name;
-  if (net?.startsWith("matic")) net = "poly";
-  else net = "eth";
-  const kid = `${net}:${chain}:${iss}`;
+export async function userCreatesToken(
+  signer: Signer,
+  chainId: number
+): Promise<Token> {
+  const now = Date.now();
+  const exp = new Date(now + 10 * 60 * 60 * 1000).toISOString(); // Default to ~10 hours
 
-  return await createToken(
-    sign,
-    { kid: kid, alg: "ETH" },
-    { iss: iss, exp: exp }
-  );
+  return await createToken(signer, {
+    chainId: chainId,
+    expirationTime: exp,
+    uri: globalThis.document?.location.origin,
+    version: "1",
+    statement: "Official Tableland SDK",
+  });
 }
 
 export async function connect(options: ConnectionOptions): Promise<Connection> {
-  const network = options.network ?? "testnet";
-  const host = options.host ?? "https://testnet.tableland.network";
+  const network = options.network ?? "goerli";
+  const host = options.host ?? "https://testnetv2.tableland.network";
+  const contract = options.contract ?? contractAddresses[network];
 
-  if (network !== "testnet" && !options.host) {
+  if (network !== "goerli" && !options.host) {
```

Comment:
updating the defaults

Reply:
So here's the question then, is network here mean to identify the tableland network, or the chain network. Since host already has the tableland network information in it, perhaps we use network to identify the chain network?

Reply:
After our chat today it occurs to me that this might be related to #22 and the signer checks for provider as well.  
TLDR; I think what we have here works well, but thinking through all the connection params again in combination with #22 could be good.

Some more details:

The SDK needs the following to enable **ALL** the method calls:
 1. An http url for the validator/gateway that the caller wants to connect to Tableland with.  Note that for everything to work this validator is going to have to support the chain in number 5 below and share the smart contract address.
 2. A signer object to use for proving the user's identity to the Validator, EVM, etc...
 3. If the caller wants to call `create` a provider object, as defined by `ethers.js`, is needed.  This is used for any direct interaction with the EVM.  At the moment this is only the create method, but I can imagine other cases like directly setting or unsetting the controller.  If we stop doing the strict checks on signer this could be optional and then error if the create method is called on a connection that doesn't have a provider.
 4. The address of the Tableland Tables smart contract deployment (registry contract) that will be shared between the SDK caller and the Validator that the SDK caller specified above.  
 5. The chainId of the EVM where the shared contract is deployed

The combination of 1, 4, and 5 is what I've been thinking about as a Tableland Network.  I could see changing the connection params to enable something like `connect({network: 'goerli-testnet'})` where the validator/gateway url is set to `https://testnetv2.tableland.network/` the contract address is set to `0xsdasfdwhatever`, and the chainId is set to `5`.  If the __optional__ provider's getNetwork method exists and doesn't match the specified Tableland Network's requirement then we would have to error out.
Sorry that's is all a lot for a cr comment.  Maybe we scope this out in #22?

Reply:
Naw this is super helpful! I'm going to finish this up tomorrow and we'll see what everyone thinks. I like this approach of only requiring a provider when explicitly needed.